ssl/s3_lib.c

```diff
@@ -4000,7 +4000,8 @@ long ssl3_ctx_ctrl(SSL_CTX *ctx, int cmd, long larg, void *parg)
                 return 0;
             }
         }
-        if (!X509v3_cache_extensions((X509 *)parg, ctx->libctx, ctx->propq)) {
+        X509_set0_libctx((X509 *)parg, ctx->libctx, ctx->propq);
+        if (!X509v3_cache_extensions((X509 *)parg)) {
```

Comment:
I think none of these X509v3_cache_extensions() calls are required any more in libssl. They were added recently because of a lack of X509_set0_libctx.

Reply:
Not sure I quite understand the original intent..
X509v3_cache_extensions() is doing a lot of checks.. Are you saying they will get done later when needed?? So they were only being done here so the libctx was known?

Reply:
Ahh ok I get it..